model/task/db.go

```diff
@@ -195,10 +198,13 @@ var (
 			"branches": []bson.M{
 				{
 					"case": bson.M{
-						"$ne": []interface{}{
-							bson.M{
-								"$size": bson.M{"$ifNull": []interface{}{"$annotation_docs", []bson.M{}}},
-							}, 0,
+						"$or": []bson.M{
+							{"$eq": []interface{}{"$" + HasAnnotationsKey, true}},
+							{"$ne": []interface{}{
+								bson.M{
+									"$size": bson.M{"$ifNull": []interface{}{"$annotation_docs", []bson.M{}}},
+								}, 0,
```

Comment:
Do we still need this now that we can use hasAnnotation to calculate the display status?

Reply:
Yes because this will still be needed for old versions in the period that we're maintaining backward compatability.